fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java

```diff
@@ -84,6 +94,78 @@ public static ConnectContext buildConnectContext() {
         return context;
     }
 
+    private static StatsConstants.AnalyzeType parseAnalyzeType(TransactionState txnState, Table table) {
+        Long loadRows = null;
+        TxnCommitAttachment attachment = txnState.getTxnCommitAttachment();
+        if (attachment instanceof LoadJobFinalOperation) {
+            EtlStatus loadingStatus = ((LoadJobFinalOperation) attachment).getLoadingStatus();
+            loadRows = loadingStatus.getLoadedRows(table.getId());
+        } else if (attachment instanceof InsertTxnCommitAttachment) {
+            loadRows = ((InsertTxnCommitAttachment) attachment).getLoadedRows();
+        } else if (attachment instanceof StreamLoadTxnCommitAttachment) {
+            loadRows = ((StreamLoadTxnCommitAttachment) attachment).getNumRowsNormal();
+        }
+        if (loadRows != null && loadRows > Config.statistic_sample_collect_rows) {
+            return StatsConstants.AnalyzeType.SAMPLE;
+        }
+        return StatsConstants.AnalyzeType.FULL;
+    }
+
+    public static void triggerCollectionOnFirstLoad(TransactionState txnState, Database db, Table table,
+                                                    boolean sync, CountedListener listener) {
+        if (!Config.enable_statistic_collect_on_first_load) {
+            listener.run();
+            return;
+        }
+        if (statisticDatabaseBlackListCheck(db.getFullName())) {
+            listener.run();
+            return;
+        }
+        // Check if it's the first load
+        if (GlobalStateMgr.getCurrentState().getAnalyzeManager().hasBasicStatsMeta(table.getId())) {
```

Comment:
why this method could express this table is first load.

Reply:
if this feature is enabled, then statistics collection will be triggered on first load, basic status meta will be created in the meanwhile.